src/main.js

```diff
@@ -19,31 +19,63 @@ userNewBody.addEventListener('input', verifyForm);
 
 ideaGallery.addEventListener('click', function(event) {
   event.preventDefault();
-  var activeStarBtn = document.querySelector('.star-active');
-  var inactiveStarBtn = document.querySelector('.star-inactive');
 
   if (event.target.className === 'delete') {
     deleteFromArray(event);
     deleteFromDOM(event);
   }
 
-  if (event.target.className === 'star-active') {
-    inactiveStarBtn.classList.remove('hide');
-    activeStarBtn.classList.add('hide');
-  }
-
   if (event.target.className === 'star-inactive') {
-    inactiveStarBtn.classList.add('hide');
-    activeStarBtn.classList.remove('hide');
+    toggleStarOnData(event);
+    toggleStarOnDOM(event);
+    console.log(savedIdeas);
+  } else if (event.target.className === 'star-active') {
+    toggleStarOffData(event);
+    toggleStarOffDOM(event);
+    console.log(savedIdeas);
   }
 })
 
+function toggleStarOnData() {
+  var clickedStar = event.target.closest('.box');
+  var starIdNumber = parseInt(clickedStar.id);
+
+  for (var i = 0; i < savedIdeas.length; i++) {
+    if (starIdNumber === savedIdeas[i].id) {
+      savedIdeas[i].star = true;
+    }
+  }
+}
+
+function toggleStarOnDOM() {
+  var starImg = event.target;
+  starImg.src = "assets/star-active.svg";
+  starImg.className = "star-active";
+}
+
+function toggleStarOffData() {
+  var clickedStar = event.target.closest('.box');
+  var starIdNumber = parseInt(clickedStar.id);
+
+  for (var i = 0; i < savedIdeas.length; i++) {
+    if (starIdNumber === savedIdeas[i].id) {
+      savedIdeas[i].star = false;
+    }
+  }
+}
+
+function toggleStarOffDOM() {
+  var starImg = event.target;
+  starImg.src = "assets/star.svg";
+  starImg.className = "star-inactive";
+}
+
 function deleteFromArray(event) {
   var boxToRemove = event.target.closest('.box');
-  var boxNumber = parseInt(boxToRemove.id);
+  var boxIndexNumber = parseInt(boxToRemove.id);
   for (var i = 0; i < savedIdeas.length; i++) {
-    if (boxNumber === savedIdeas[i].id) {
-      savedIdeas.splice(savedIdeas.indexOf(boxNumber));
+    if (boxIndexNumber === savedIdeas[i].id) {
+      savedIdeas.splice(savedIdeas.indexOf(savedIdeas[i]));
```

Comment:
This was a small bug fix from before. It allows the deletion to come from the savedIdeas array by way of the for loop.